obspy/core/inventory/response.py

```diff
@@ -598,7 +640,17 @@ def get_response(self, frequencies):
         # evalresp does this and thus so do we.
         if self.cf_transfer_function_type != 'DIGITAL':
             amp *= self.stage_gain / gain_freq_amp
-        final_resp = np.empty_like(resp)
+
+        # If "fast", then interpolate the amplitude and phase onto the
+        # originally requested frequencies.
+        if len(frequencies) > 10000 and fast:
+            amp = scipy.interpolate.InterpolatedUnivariateSpline(
+                    resp_frequencies, amp, k=2)(frequencies)
+            phase = scipy.interpolate.InterpolatedUnivariateSpline(
+                    resp_frequencies, phase, k=2)(frequencies)
+            final_resp = np.zeros_like(frequencies) + 0j
```

Comment:
I think this is duplicated, better define a helper function for the interpolation part

```python
def _interpolate(resp_frequencies, amp, phase, frequencies):
    ....
```

Reply:
In https://github.com/amkearns-usgs/obspy/pull/5 I have added a helper function  `_check_response_interpolation` to do the check whether interpolation is necessary, and to provide the log-spaced-values. I didn't include the interpolation-call itself into the helper function, because it's only twice used on both phase and amplitude, and once on only the amplitude - so I think in the current version saves us the most lines of code and is easiest to read. But let me know if you prefer it differently.

Reply:
I only saw know that you do it separately, so I would still refactor it, duplicated code is always good to avoid (function naming is just a proposol, other name is fine)

```python
def _interpolate(resp_frequencies, quantity, frequencies):
    ...

amp = _interpolate(resp_frequencies, amp, frequencies)
phase = _interpolate(resp_frequencies, phase, frequencies)
```

Also, I see linspace used there.. so linear spacing? Should be logspace instead?

Reply:
Ok, no problem, added the interpolate-helper function as well.
Regarding the linspace, thanks for catching that - it shouldn't have been there and must have gotten back in when solving some merge conflicts..

I did some more tests for the linspace vs logspace-issue. As expected, for `FIRResponseStage` the logspace introduces a bit more of a difference, while for `PolesZerosResponseStage`, using linspace introduces some more - but they are within very reasonable limits.
Even though I did quite some tests, it's probably still good to also test the interpolation on a larger set of responses / seismograms. Sometimes I got extremely small relative differences after correcting with calculated vs. interpolated responses (e.g., 10^-13), and sometimes I got still small, but bigger differences (relative error: ~10^-5). May be related to values very close to 0 (--> waterlevel etc.).
It appears to me that

Reply:
Without having looked into it myself, if both linear and log-spacing isn't ideal.. there are other spacing strategies out there as well. I don't have a good resource, but I think these matplotlib axes scale graphs give a good idea (e.g. something like logit, which looks like log spacing but with dense points at both ends): https://matplotlib.org/3.1.0/gallery/scales/scales.html